### PR TITLE
Enchancements for AWS S3 plugin

### LINKED ISSUE #3078: Enhancements to AmazonS3 plugin
## Summary
1. Return expiry date when signed URL is generated - for List and Create actions. 
2. In List action, make url expiry field dependent on 'Generate signed url' toggle. 
3. Rename labels.

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Return expiry date when signed URL is generated - for List and Create actions.
- In List action, make url expiry field dependent on 'Generate signed url' toggle.
- Rename labels.
- Change expiry duration input to minutes instead of milliseconds.
- Set default value for expiry duration.
- Update TCs.

Fixes #3078

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
https://user-images.githubusercontent.com/1757421/108202107-a4b66b80-7146-11eb-8d9d-c533a56e375f.mov

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 1]
```diff
@@ -425,59 +424,68 @@ String readFile(AmazonS3 connection, String bucketName, String path, Boolean enc
...
+                                fileInfo.put("urlExpiryDate", expiryDateTimeString);
+                                ((ArrayList<Object>)actionResult).add(fileInfo);
                             }
-
-                            actionResult = Map.of("files", listOfFilesAndUrls);
                         }
                         else {
-                            actionResult = Map.of("files", listOfFiles);
+                            actionResult = new ArrayList<Object>();
+                            for(int i=0; i<listOfFiles.size(); i++) {
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:438]: Please format the entire file. This ensures that the style is consistent across the board.

Comment (sumitsum) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:438]: @mohanarpit you mean display result as array even for read, create and delete actions ? read and delete return only a single value as response and create action only returns two values in response. @Nikhil-Nandagopal need your opinion here.

Comment (mohanarpit) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:438]: No. I meant code formatting for indentation, spaces etc. Nothing to do with the structure of the response.

Comment (sumitsum) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:438]: Done.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**